### DISCUSSION TITLE
All query parameters should be loaded into the API client (disabled if not required)

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [scalar/scalar#731] [Ideas]
Hi,
It would be super useful to have the query section pre-populated with all the querystrings defined in the openapi/swagger schema.

I've seen it happen for the required ones, but I think they should all be there, maybe disabled so that users can populate and activate the one they prefer.

What do you think?

This below is just an example:
<details><summary>Details</summary>
<p>

```json
{
   "openapi":"3.0.3",
   "info":{
      "title":"example",
      "description":"",
      "version":"1.0.0"
   },
   "paths":{
      "/":{
         "get":{
            "parameters":[
               {
                  "schema":{
                     "type":"string"
                  },
                  "in":"query",
                  "name":"required",
                  "required":true,
                  "description":"this is pre-populated"
               },
               {
                  "schema":{
                     "type":"string"
                  },
                  "in":"query",
                  "name":"not required",
                  "required":false,
                  "description":"this is not shown"
               }
            ]
         }
      }
   }
}
```

![image](https://github.com/scalar/scalar/assets/39826608/5c3e63c4-7e58-428c-8200-7d8d3013351c)

</p>
</details>

**Comment by marclave** [comment]
Appreciate your comment!

We removed that in our integration of fastify and other frameworks where there was lots of optional params, however this was not the best solution :) 

I think we should add back the optional params unchecked

if there are less than 5 we show them in the card normally, however if there are more than 5 we have a "show more" button that by default hides the additional optional queries to then open.

This solves both problems! @cameronrohani and I will mock something up and hopefully get a PR in soon

**Reply by hanspagel** [reply]
Yep, great suggestion.
